Add artifact transforms to release notes (#9485)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

`./gradlew wrapper --gradle-version=@version@`

## Transforming dependency artifacts on resolution

> A dependency’s artifacts can take many forms, some of which will be published but others may not be.

Reviewer: This sentence is confusing. Why don't we just talk about a simple Java module that gets published in the usual (unobfuscated) way, and that you want all your dependencies (including this one) obfuscated, so you employ an artifact transform to do it.

Author: Can you make a suggestion? I don't find this confusing, since for me, that is what artifact transforms are about: convert one form of an artifact to another one since it has not been published in the desired form.

Reviewer: ```suggestion
A dependency’s artifacts can take many forms, and sometimes you might need them in a form that is not readily available in a published form.
```

> A dependency’s artifacts can take many forms, and sometimes you might need them in a form that is not readily available in a published form.

Reviewer: Damn, I missed that we now use `form` thrice in the sentence!
```suggestion
A dependency’s artifacts can take many forms, and sometimes you might need one that is not readily available.
```

As an example, imagine you have a dependency on a Java module.
Its producer publishes a normal JAR file and an obfuscated JAR file to a binary repository.
As a consumer, you want to use the normal JAR for development.
But for production, you want to use an obfuscated version of the JAR rather than the version that’s published.

Let’s say you want to get hold of the obfuscated JAR, but it’s not available in any repository.
Why not just retrieve the un-obfuscated JAR and obfuscate it yourself?

Gradle now allows you to register an _artifact transform_ to do just that, by hooking into the dependency management resolution engine.
You can specify that whenever an obfuscated JAR is requested but can’t be found, Gradle should run an artifact transform that performs the obfuscation and makes the resulting artifact available to the build.

> The build won’t even be aware that the artifact transform has run.

Reviewer: ```suggestion
The build won’t even be aware that the artifact was transformed.
```
